src/main/java/com/contentful/java/cda/interceptor/ContentfulUserAgentHeaderInterceptor.java

```diff
@@ -0,0 +1,299 @@
+package com.contentful.java.cda.interceptor;
+
+import java.util.LinkedHashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.lang.String.format;
+import static java.util.Locale.getDefault;
+import static java.util.regex.Pattern.compile;
+
+/**
+ * This header interceptor will add contentful relevant information to the custom user agent header.
+ */
+public class ContentfulUserAgentHeaderInterceptor extends HeaderInterceptor {
+  public static final String HEADER_NAME = "X-Contentful-User-Agent";
+
+  /**
+   * A section of values used to fill out the Contentful custom HTTP header.
+   */
+  public static class Section {
+    /**
+     * Defines a version of this section.
+     */
+    public static class Version {
+
+      private static final String VERSION_REGEX = "^([0-9]+).([0-9]+).([0-9]+)(.*)?$";
+      private static final Pattern VERSION_PATTERN = compile(VERSION_REGEX);
+      private static final String STABILITY_REGEX = "^([A-Z]+[0-9]*).*";
+      private static final Pattern STABILITY_PATTERN = compile(STABILITY_REGEX);
+
+      /**
+       * Convert a version into a Semver and Contentful conform version number.
+       *
+       * @param version a String like '1.2.3-dev' to parse.
+       * @return a version of (1, 2, 3, dev)
+       * @throws NumberFormatException    if major, minor or patch are no numbers
+       * @throws IllegalArgumentException if numbers are negative.
+       * @throws IllegalArgumentException if stability cannot be read.
+       */
+      public static Version parse(String version) {
+        final Matcher matcher = VERSION_PATTERN.matcher(version);
+
+        if (matcher.find() && matcher.groupCount() == 4) {
+          // ignore first full matching result
+          final int major = Integer.parseInt(matcher.group(1));
+          final int minor = Integer.parseInt(matcher.group(2));
+          final int patch = Integer.parseInt(matcher.group(3));
+
+          if (major < 0) {
+            throw new IllegalArgumentException("Major version number cannot be negative!");
+          }
+          if (minor < 0) {
+            throw new IllegalArgumentException("Minor version number cannot be negative!");
+          }
+          if (patch < 0) {
+            throw new IllegalArgumentException("Patch version number cannot be negative!");
+          }
+
+          final String stability = parseStability(matcher.group(4));
+          return new Version(major, minor, patch, stability);
+        } else {
+          throw new IllegalArgumentException("Could not parse version: " + version);
+        }
+      }
+
+      private static String parseStability(String stability) {
```

Comment:
mind leaving a developer comment so that future devs know this refers to modifiers such as `-beta` `-RC` and `-alpha`? "stability" wasn't clear to me just from the title